src/python/pants/backend/python/lint/pylint/rules_integration_test.py

```diff
@@ -0,0 +1,160 @@
+    return self.request_single_product(
+      LintResult, Params(target, create_options_bootstrapper(args=args)),
+    )
+
+  def test_single_passing_source(self) -> None:
+    self.create_python_library()
+    self.write_file(self.good_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_single_failing_source(self) -> None:
+    self.create_python_library()
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source])
+    assert result.exit_code == 16
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  def test_mixed_sources(self) -> None:
+    self.create_python_library()
+    self.write_file(self.good_source)
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.good_source, self.bad_source])
+    assert result.exit_code == 16
+    assert "good.py" not in result.stdout
+    assert "bad.py:1:0: C0114" in result.stdout
+
+  def test_precise_file_args(self) -> None:
+    self.create_python_library()
+    self.write_file(self.good_source)
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+  
+  @pytest.mark.skip(reason="Get config file creation to work with options parsing")
+  def test_respects_config_file(self) -> None:
+    self.create_python_library()
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source], config="[pylint]\ndisable = C0114\n")
+    assert result.exit_code == 0
+    assert result.stdout.strip() == ""
+
+  def test_respects_passthrough_args(self) -> None:
+    self.create_python_library()
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source], passthrough_args="--disable=C0114")
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_includes_direct_dependencies(self) -> None:
+    self.create_python_library(name="library")
+    self.create_python_library(
+      name="dependency", sources=["dependency.py"], dependencies=[":library"],
+    )
+    self.write_file(
+      FileContent(
+        path=f"{self.source_root}/dependency.py",
+        content=dedent(
+          """\
+          from nonexistent import erroneous
+          VAR = 42
+          """
+        ).encode(),
+      )
+    )
+    source = FileContent(
+      path=f"{self.source_root}/test_dependency.py",
+      content=dedent(
+        """\
+        '''Docstring'''
+        from dependency import VAR
+        assert VAR == 42
+        """
+      ).encode(),
+    )
+    self.create_python_library(sources=["test_dependency.py"], dependencies=[":dependency"])
+    self.write_file(source)
+    result = self.run_pylint([source])
+    assert result.exit_code == 0
```

Comment:
I personally might also consider running this with the same inputs, but modifying the file so that the import fails (which I would *think* is what `from nonexistent import erroneous` would do?). My concern when just testing success is that (for example) if later on someone makes a mistake updating this test and provides it zero source files, it might silently start reading 10/10 no matter what. We're not testing that the control is correctly setup here, I think.